### PR TITLE
add docstrings to the functions of the icinga class

### PR DESCRIPTION
**Pull request by rndmh3ro**

### CODE REVIEW

File: plugins/module_utils/icinga.py [hunk 1]
```diff
@@ -65,19 +89,51 @@ def exists(self, find_by="name"):
...
 
     def create(self):
+        """
+        Create the object in the director and return the result of the api-call.
+
+        Parameters:
+            none
+
+        Returns:
+            the results of the api-call
```
Comment (schurzi) [inline plugins/module_utils/icinga.py:99]:
```suggestion
            the result of the api-call
```

File: plugins/module_utils/icinga.py [hunk 2]
```diff
@@ -52,6 +67,15 @@ def call_url(self, path, data="", method="GET"):
         return {"code": info["status"], "data": content, "error": error}
 
     def exists(self, find_by="name"):
+        """
+        Check if the object already exists in the director.
+
+        Parameters:
+            find_by:
```
Comment (schurzi) [inline plugins/module_utils/icinga.py:74]: description missing

File: plugins/module_utils/icinga.py [hunk 3]
```diff
@@ -24,6 +26,19 @@ def __init__(self, module, path, data):
         self.object_id = None
 
     def call_url(self, path, data="", method="GET"):
+        """
+        Execute the request against the API with the provided arguments and return json.
+
+        Parameters:
+            path: the path of the api to call
+            data: the module params passed to the api
+            method: the method to run against the api.
```
Comment (schurzi) [inline plugins/module_utils/icinga.py:35]: should we also include datatype and defaults?

e.g.:
```
method : str, default 'GET'
    the method to run against the api
```

**Review by schurzi — APPROVED** [review]

File: plugins/module_utils/icinga.py (hunk 3 above)
Comment (rndmh3ro) [reply plugins/module_utils/icinga.py]: I added them everywhere.

**Pull request merged**